bigtable/tests/data_integration_test.cc

```diff
@@ -12,84 +12,177 @@
-  table.Apply(std::move(mutation));
-  std::cout << "row-key-0 mutated successfully" << std::endl;
+  (void)::testing::AddGlobalTestEnvironment(
+      new ::bigtable::testing::TableTestEnvironment(project_id, instance_id));
 
-  mutation = bigtable::SingleRowMutation("row-key-1");
-  mutation.emplace_back(bigtable::SetCell(family, "col0", 0, "value-1-0"));
-  mutation.emplace_back(bigtable::SetCell(family, "col1", 0, "value-1-1"));
-  table.Apply(std::move(mutation));
-  std::cout << "row-key-1 mutated successfully" << std::endl;
-
-  bigtable::BulkMutation bulk{
-      bigtable::SingleRowMutation("row-key-2",
-                                  {bigtable::SetCell(family, "c0", 0, "v0"),
-                                   bigtable::SetCell(family, "c1", 0, "v1")}),
-      bigtable::SingleRowMutation("row-key-3",
-                                  {bigtable::SetCell(family, "c0", 0, "v2"),
-                                   bigtable::SetCell(family, "c1", 0, "v3")}),
-  };
-  table.BulkApply(std::move(bulk));
-  std::cout << "bulk mutation successful" << std::endl;
+  return RUN_ALL_TESTS();
 
-  return 0;
 } catch (bigtable::PermanentMutationFailure const& ex) {
   std::cerr << "bigtable::PermanentMutationFailure raised: " << ex.what()
             << " - " << ex.status().error_message() << " ["
             << ex.status().error_code()
             << "], details=" << ex.status().error_details() << std::endl;
+  int count = 0;
+  for (auto const& failure : ex.failures()) {
+    std::cerr << "failure[" << count++
+              << "] {key=" << failure.mutation().row_key() << "}" << std::endl;
+  }
   return 1;
 } catch (std::exception const& ex) {
   std::cerr << "Standard exception raised: " << ex.what() << std::endl;
   return 1;
+} catch (...) {
+  std::cerr << "Unknown exception raised." << std::endl;
+  return 1;
+}
+
+namespace bigtable {
+namespace testing {
+
+// lets insert a single row at a time
+void DataIntegrationTest::Apply(bigtable::Table& table, std::string row_key,
+                                std::vector<bigtable::Cell> const& cells) {
+  auto mutation = bigtable::SingleRowMutation(row_key);
+  for (auto const& cell : cells) {
+    mutation.emplace_back(bigtable::SetCell(
+        static_cast<std::string>(cell.family_name()),
+        static_cast<std::string>(cell.column_qualifier()), cell.timestamp(),
+        static_cast<std::string>(cell.value())));
+  }
+  table.Apply(std::move(mutation));
 }
+
+// Insert multiple rows
+void DataIntegrationTest::BulkApply(bigtable::Table& table,
+                                    std::vector<bigtable::Cell> const& cells) {
+  std::map<std::string, bigtable::SingleRowMutation> mutations;
+  for (auto const& cell : cells) {
+    std::string key = static_cast<std::string>(cell.row_key());
+    auto inserted = mutations.emplace(key, bigtable::SingleRowMutation(key));
+    inserted.first->second.emplace_back(bigtable::SetCell(
+        static_cast<std::string>(cell.family_name()),
```

Comment:
I think the `static_cast<std::string>` is no longer needed after #216

Reply:
fixed